File: Backend/TagsInstrumentsService/src/main/java/io/github/juniorcorzo/tagsinstrumentsservice/orchestrator/dtos/UnitProcessDTO.java
package io.github.juniorcorzo.tagsinstrumentsservice.orchestrator.dtos;

import com.fasterxml.jackson.annotation.JsonProperty;
import com.fasterxml.jackson.annotation.JsonTypeName;

@JsonTypeName("unitProcess")
public record UnitProcessDTO(
        @JsonProperty("id")
        String id,
        String name,
        String description,
        CampDTO camp
) {

}

record CampDTO(String name, Location location) {

}

record Location(
        String municipality,
        String department,
        String country,
        String[] coordinates
){}

File: Backend/unit-process-service/src/unitProcess/dtos/unit-process.dto.ts
import { ObjectId } from "mongodb";
import { CampDTO } from "src/camp/dtos/camp.dto";

interface IUnitProcessDTO {
  id: ObjectId;
  name: string;
  description: string;
  camp: CampDTO;
}

export class UnitProcessDTO implements IUnitProcessDTO {
  readonly id: ObjectId;
  readonly name: string;
  readonly camp: CampDTO;
  readonly description: string;

  constructor(id: ObjectId, name: string, description: string, camp: CampDTO) {
    this.id = id;
    this.name = name;
    this.description = description;
    this.camp = camp;
  }
}
